### PR TITLE
fix: make `ForumChannel.create_thread` not require content

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

The `content` parameter isn't required when creating threads in forum channels, similar to `Messageable.send` at least one of `content`/`embeds`/... has to be set.

Depends on #732 somewhat, but the documentation change could be moved to #732 instead and this PR could be merged first.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: changelog/739.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Change :func:`ForumChannel.create_thread` to not require a ``content`` parameter.
```
Comment (onerandomusername) [inline changelog/739.bugfix.rst:1]:
```suggestion
Change :func:`ForumChannel.create_thread` to not require the ``content`` parameter.
```

Comment (shiftinv) [reply changelog/739.bugfix.rst:1]:
```suggestion
Change :func:`ForumChannel.create_thread` to not require the ``content`` parameter to be provided.
```
like that maybe?

Comment (onerandomusername) [reply changelog/739.bugfix.rst:1]: Hm, we should probably allude to all of the required parameters in some form.

Comment (shiftinv) [reply changelog/739.bugfix.rst]: https://github.com/DisnakeDev/disnake/pull/739/commits/8b510731db958ee71c6bee601be9da64ecd3fbd7

**Review by onerandomusername — APPROVED** [review]

**Comment by onerandomusername** [comment]
While I just approved, this is actually a feature, not a bugfix, IMO

**Comment by shiftinv** [comment]
> While I just approved, this is actually a feature, not a bugfix, IMO

It shouldn't have been required from the start, this change is mostly unrelated to the recent components-only feature

**Pull request merged**